Lab10/closedunidirectionallinkedlist.py

```diff
@@ -0,0 +1,198 @@
+from device import Device
+
+
+class Node:
+    def __init__(self, data: Device = None) -> None:
+        self.data: Device = data
+        self.next: Node = None
+
+
+class ClosedUnidirectionalLinkedList:
+    def __init__(self) -> None:
+        self.head: Node = Node(None)
+        self.tail: Node = Node(None)
+        self.head.next = self.tail
+        self.tail.next = self.head
+
+    # add -> add some element type Device to list
+    def add(self, data: Device) -> None:
+        new_node: Node = Node(data)
+        if self.head.data is None:
+            self.head = new_node
+            self.tail = new_node
+            new_node.next = self.head
+        else:
+            self.tail.next = new_node
+            self.tail = new_node
+            self.tail.next = self.head
+
+    # print_list -> print all elements of list
+    def print_list(self) -> None:
+        current: Node = self.head
+        if self.head.data is None:
+            print("List is empty!")
+        else:
+            print("Nodes of the circular linked list: ")
+            print(current.data, end=''),
+            while current.next != self.head:
+                current = current.next
+                print(current.data, end='')
+
+    # get -> return some element
+    def get(self, index: int) -> Device:
+        k: int = 0
+        current: Node = self.head
+        if self.head.data is None:
+            raise IndexError("List index out of range")
+        else:
+            if k == index:
+                return current.data
+            else:
+                while current.next != self.head:
+                    k += 1
+                    current = current.next
+                    if k == index:
+                        return current.data
+                    elif k > index:
+                        raise IndexError("List index out of range")
+
+    # get_information_about_element -> get information about data from element
+    def get_information_about_element(self, index: int) -> None:
+        k: int = 0
+        current: Node = self.head
+        if self.head.data is None:
+            raise IndexError("List index out of range")
+        else:
+            if k == index:
+                print(current.data, end='')
+                return
+            else:
+                while current.next != self.head:
+                    k += 1
+                    current = current.next
+                    if k == index:
+                        print(current.data, end='')
+                        return
+                    elif k > index:
+                        raise IndexError("List index out of range")
+
+    # output_of_devices_with_the_set_limit -> output information about device, which have same limit of measurement
+    def output_of_devices_with_the_set_limit(self, limit_of_measurement: int) -> None:
+        current: Node = self.head
+        if self.head.data is None:
+            raise NotImplemented("List is empty!")
```

Comment:
It could be a custom exception as well, as it's not really NotImplemented. NotImplemented is used when your method does not support some data type yet, or the whole functionality is not implemented yet (but will be soon), and you want to indicate that to your colleagues. E. g., the front-end team needs to call an endpoint for their new web page, but the endpoint is not implemented yet, because it's one of your future tasks. They can call the endpoint here at this moment, but the implementation will be delivered later.